kubeflow/kubeflow
# Adds generalized webhook

- integrated the existing gcp-admission-webhook with an implementation of [podpreset](https://github.com/jpeeler/podpreset-crd)
- resolves #2844
- related to https://github.com/kubeflow/manifests/pull/33
- please [read here](https://github.com/zabbasi/kubeflow/blob/webhook11/components/admission-webhook/README.md) to see how it works.
- tested manually on a kubeflow cluster
   *  create podpreset in different namespaces with various specs
   * create pods in different namespace and check that the expected podpresets is applied to the pods





---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/kubeflow/kubeflow/3202)

/assign @lluunn
